libs/grout/README.md

```diff
@@ -0,0 +1,160 @@
+# Grout
+
+A library to create type-safe RPC glue code.
+
+Grout lets you define a server API as a plain TS object with async methods.
+Based on this API definition, Grout will create Express route handlers on the
+server side and a type-safe client on the client side, abstracting away the
+details of the underlying HTTP requests.
+
+## Installation
+
+Grout must be installed as a dependency in both your client and server projects.
+
+## Usage
+
+### Server
+
+```ts
+import * as grout from '@votingworks/grout';
+
+interface Person {
+  name: string;
+  age: number;
+}
+
+// Create an API definition by building an object with async methods.
+const api = grout.createApi({
+  // Each method should return a Promise. The data returned should be simple TS
+  // values - not fancy class instances. Values that can be easily serialized to
+  // JSON. More details on this below.
+  async getAllPeople(): Promise<Person[]> {
+    return store.getAllPeople(); // Assume this accesses the database, etc.
+  },
+
+  // Methods can take input, but it must be packaged into a single object.
+  // Think of it like using named parameters, or React component props.
+  async getPersonByName(input: { name: string }): Promise<Person | undefined> {
+    return store.getPersonByName(input.name);
+  },
+
+  // For known errors (e.g. invalid input), use a Result type so that the client
+  // forced to handle them explicitly.
```

Comment:
For my understanding, what exactly do you mean when you say "forced to handle them explicitly"? Do you mean that TS will warn you if you don't (I don't think that's the case)? Or that you can't rely on some global error handler like the react-query error handler?

Reply:
Since `Result` is a wrapper and is always either `ok(someVal)` or `err(someErr)`, you always have to check which one it is and unwrap to get the result. This is basically a strongly recommended pattern that @eventualbuddha came up with so that we can't forget to handle known errors (rather than use exceptions/a global error handler).

From a UX perspective, the idea is to surface known errors to users near the point of action that induced the error and reserve a global error handler for unexpected errors.

Reply:
:100: Thank you for the explanation!

Reply:
Thinking through this more though, it's actually still very possible to forget to handle the `Err` case with a mutation that returns `Result<void, Error>`, since we don't need to check the `Ok` case to get an output value. @eventualbuddha do you have any thoughts about how we might prevent this kind of mistake?

Reply:
We actually have an ESLint rule for this but it's broken! I'm fixing it now. It'll look like this when it's working:

<img width="699" alt="image" src="https://user-images.githubusercontent.com/1938/206543326-2423da93-9690-476e-8c85-0849289f1fd2.png">

Reply:
Awesome!